docs: database structure (#80)
Alzymologist/Kalatori-backend: docs/DATABASE.md

> Plan is to update the database scheme in a way that it will support the requirements we have as for the API specs and additional improvements of the deamon.

Reviewer: are these things derived from API? Please indicate if something is different from types defined there, ideally these should be as similar as possible, but deviations are ok where they make sense

## Tables
### Orders (`orders`)
- order - String: order identifier provided by the frontend 
- payment_status - Enum: (pending|paid|timed_out). 

> - withdrawal_status - Enum: (waiting|failed|completed|none). 

Reviewer: we might need "forced" status as well

Author: Addressed in [c69aa99](https://github.com/Alzymologist/Kalatori-backend/pull/80/commits/c69aa99b6747242f1a30f2348185c35d147486d8)

- amount - Float: Order amount 
- currency - String: Currency ticker ("DOT"|"USDC"|...). 
- callback: String: Callback url for frontend order status update 
- payment_account: Derived address for this order. 
- recipient: Address that will receive the payout once the order is fulfilled. 
- death: Expiry timestamp for the order.

### Transactions (`transactions`)
- order - String: order id to link transaction to order
- chain - String: identifier for the chain where transaction occured
- block_nmber - Integer: Block number where the transaction is recorded.

> - position_in_block - Integer: Position of the transaction within the block. 

Reviewer: we might also (later?) want to store the block for which transaction was submitted. It is included into saved transaction bytes though, so maybe not

Plan is to update the database scheme in a way that it will support the requirements we have as for the API specs and additional improvements of the deamon.

## Tables
### Orders (`orders`)

> - order - String: order identifier provided by the frontend 

Author: ```suggestion
- order - String: order identifier provided by the frontend 
- chain - H64: chain name where order is planned to be repaid
```

- payment_status - Enum: (pending|paid|timed_out). 
- withdrawal_status - Enum: (waiting|failed|completed|forced|none). 

> - amount - Float: Order amount 

Author: ```suggestion
- amount - u128: Order amount 
```

- currency - String: Currency ticker ("DOT"|"USDC"|...). 
- callback: String: Callback url for frontend order status update 

> - payment_account: String: Derived address for this order. 

Author: ```suggestion
- payment_account: [u8; 32]: Derived address for this order. 
```

> - recipient: String: Address that will receive the payout once the order is fulfilled. 

Author: ```suggestion
- recipient: [u8; 32]: Address that will receive the payout once the order is fulfilled. 
```

- message: String|null: Optional parameter for failed orders.
- payment_page: String|null: Optional parameter for the frontend to redirect to a payment page.
- redirect_url: String|null: Optional parameter for the frontend to redirect once the order is repaid.

> - death: Timestamp: Expiry timestamp for the order.

Author: ```suggestion
- death: u32: Expiry timestamp for the order.
```

### Transactions (`transactions`)

> - order - String: order id to link transaction to order

Author: ```suggestion
- transaction_id - unique id generated by us to allow linking transaction to order
- order - String: order id to link transaction to order
```

- chain - String: identifier for the chain where transaction occurred
- block_number - Integer: Block number where the transaction is recorded.
- position_in_block - Integer: Position of the transaction within the block. 
- timestamp - Timestamp: Timestamp of the transaction. 
- transaction_bytes - String: Raw transaction data. 
- sender - String: Address sending the transaction. 
- recipient - String: Address receiving the transaction. 
- amount - Float: transaction amount
- currency: String: Transaction currency 

> - status - Enum: Transaction status (pending|finalized|failed).

Author: ```suggestion
- type - Enum: Transaction type (payment|withdrawal) to distinguish between internal (withdrawal) and external (payment) transactions
- status - Enum: Transaction status (pending|finalized|failed).
```
